src/modules/formBuilder/components/formTree/useUpdateFormStructure.tsx

```diff
@@ -32,24 +39,20 @@ export default function useReorderItem(
   // grandParentArrayPath:  item        (parentIndex=3)
 
   const itemPath = itemIdMap[itemId];
-  if (!itemPath) throw new Error(`No itemPath found for linkId ${itemId}`);
   const { parentPath: parentArrayPath, index: thisIndex } =
     getPathContext(itemPath);
   const parentItemId = parentArrayPath.replace(/\.item$/, '');
   const { parentPath: grandParentArrayPath, index: parentIndex } =
     getPathContext(parentItemId);
 
   // RHF swap is used for swapping items within an array
-  const { swap } = useFieldArray({ control, name: parentArrayPath });
+  const { swap, remove } = useFieldArray({ control, name: parentArrayPath });
 
   const thisLayer = get(values, parentArrayPath);
-  if (!thisLayer)
```

Comment:
Can you add a comment indicating that `thisLayer` could be undefined if an item was just deleted